tests/collections/asr/test_asr_datasets.py

```diff
@@ -406,3 +406,168 @@ def test_dali_char_vs_ref_dataset(self, test_data_dir):
                 err = np.abs(a - b)
                 assert np.mean(err) < 0.0001
                 assert np.max(err) < 0.01
+
+    @pytest.mark.skipif(not HAVE_DALI, reason="NVIDIA DALI is not installed or incompatible version")
+    @pytest.mark.unit
+    def test_tarred_dali_char_dataset(self, test_data_dir):
+        manifest_path = os.path.abspath(os.path.join(test_data_dir, 'asr/tarred_an4/tarred_audio_manifest.json'))
+        audio_tar_filepaths = [
+            os.path.abspath(os.path.join(test_data_dir, f'asr/tarred_an4/audio_{idx}.tar')) for idx in range(2)
+        ]
+        audio_tar_index_filepaths = [
+            os.path.abspath(os.path.join(test_data_dir, f'asr/tarred_an4/dali_index/audio_{idx}.index'))
+            for idx in range(2)
+        ]
+
+        batch_size = 8
+        device = 'gpu' if torch.cuda.is_available() else 'cpu'
+        texts = []
+
+        with tempfile.NamedTemporaryFile(mode='w', encoding='utf-8') as f:
+            num_samples = 0
+            with open(manifest_path, 'r') as m:
+                for line in m:
+                    num_samples += 1
```

Comment:
How about just:
```suggestion
            num_samples = sum(1 for line in open(manifest_path, 'r'))
```

Reply:
I thought of doing @jantonguirao style, but both above options don't explicitly close the file afterwords. So I used the explicit form instead.

Reply:
so how about:
```
with open(manifest_path, 'r') as m:
    num_samples = m.count('\n')
```
?

Reply:
Nice and simple. Cool.